lib/workers/repository/process/lookup/index.ts

```diff
@@ -389,6 +389,10 @@ export async function lookupUpdates(
           update.newVersion = versioning.valueToVersion(update.newVersion!);
         }
       }
+      if (res.registryUrl) {
+        config.registryUrls = [res.registryUrl];
+      }
```

Comment:
I'm not sure of this change. Could this lead to undesirable side-effects?

Reply:
The full config object is afterwards only used for the digest lookup and not used/passed to other functions later in the method. So if, it could only have an side-effect on the digest lookup.

For the digest lookup explicitly setting the registryUrl should be side-effect free. As this is the registry, which is determined earlier (or most cases it is unset anyway and the if-block is not triggered at all).

Reply:
moved the github-tags stuff out to #24192 and will repurpose this PR for the datasource change.

Reply:
reduced this PR to only the changes to the datasource.

Reply:
Maybe, `.push` or `.unshift` to the existing `registryUrls`?

```suggestion
      if (res.registryUrl) {
        config.registryUrls ??= [];
        config.registyUrls.push(res.registryUrl);
      }
```

Reply:
`unshift` would work. Later anyway only the first registry is used, so I though we could set here one registry as well. 

https://github.com/renovatebot/renovate/blob/17bf7d7ed9626c359f6f1e74a7de72f55f024da0/lib/modules/datasource/index.ts#L396-L401

Unshifting to an array containing `["github.enteprrise.com", "github.com"]` would result in `["github.com", "github.enteprrise.com", "github.com"]`:
<img width="702" alt="image" src="https://github.com/renovatebot/renovate/assets/16856448/3f34a4ec-edb0-4e21-982e-f29cf6960b89">

Reply:
Okay, just wanted to be as conservative on changes in this function as possible